Checkout: Thank You: Move URLs to `lib/url/support`

drewblaisdell opened the pull request:
Mentioned by @stephanethomas elsewhere:

> We should use the same format for any url displayed in the Thank You page (e.g. protocole relative urls with //). As mentioned by @tugdualdekerviler2, we can for example use the client/lib/url/support.js for links to support pages.

I didn't use protocol-relative URLs, instead following the pattern in `lib/url/support`, but we should talk about whether that needs to change. I don't see a benefit to linking to `http://jetpack.me` when we're running Calypso locally, as it seems that `https://jetpack.me` should be preferred regardless of the local protocol, but I may be missing something.

**Testing**
We should assert that all of the links from the buttons on the thank you pages of the following products work as expected:

- Domain mapping
- Domain registration
- Google Apps
- Premium plan
- Chargeback item (which we can test by updating [this line](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/upgrades/checkout-thank-you/index.jsx#L174) to `return [ ChargebackDetails ];`

We should also assert that the links in the footer work when checking out with a Jetpack plan and any product that is not a Jetpack plan.

- [x] Code review
- [x] Product review

Tug commented on client/my-sites/upgrades/checkout-thank-you/domain-mapping-details.jsx:
```diff
@@ -9,6 +9,7 @@ import React from 'react';
 import { getDomainManagementUrl } from './utils';
 import { getPrimaryDomain, isSubdomain } from 'lib/domains';
 import { isPlan } from 'lib/products-values';
+import { MAP_EXISTING_DOMAIN, MAP_SUBDOMAIN } from 'lib/url/support';
```
I'm not a fan of declaring global constants here even if they are limited to this module. I'd rather have `import support from 'lib/url/support';` and use `support.MY_CONST` everywhere so we understand immediately that this constant is related to the url support lib. Those kind of small conventions can really help us once we have multiple libs declaring constants imported in the same module I think

scruffian replied:
:+1:

drewblaisdell replied:
:+1:

Tug commented on client/my-sites/upgrades/checkout-thank-you/google-apps-details.jsx:
```diff
@@ -7,6 +7,7 @@ import React from 'react';
  * Internal dependencies
  */
 import i18n from 'lib/mixins/i18n';
+import { GOOGLE_APPS_LEARNING_CENTER } from 'lib/support/url';
```
Isn't it `lib/url/support` ?

Tug commented:
I added a small comment on how we import `lib/url/support`.
I agree that protocol-relative urls are no good. If we decide that the "repository" for support documentation is at "https://support.wordpress.com", I don't think it matters on which protocol we are at the moment.

Tug commented:
:+1:

fabianapsimoes commented:
Product :+1:

stephanethomas commented on client/lib/url/support.js:
```diff
@@ -24,10 +26,13 @@ export default {
 	IMPORT: `${root}/import`,
 	INSTAGRAM_WIDGET: `${root}/instagram/instagram-widget`,
+	JETPACK_SUPPORT: `${jetpackRoot}/support`,
+	JETPACK_CONTACT_SUPPORT: `${jetpackRoot}/contact-support`,
```
We should probably add a trailing slash to these urls.

drewblaisdell replied:
Good catch.

stephanethomas commented:
The code looks good so far, I really like what I see. We should however update all urls, not only the ones for the support site. For example we shouldn't hardcode urls to the [post editor](https://github.com/Automattic/wp-calypso/blob/cc6ff97a0453df6b473f37a3747e647930847390/client/lib/paths/index.js#L21) nor the [plans](https://github.com/Automattic/wp-calypso/blob/cc6ff97a0453df6b473f37a3747e647930847390/client/my-sites/plans/paths.js#L5) pages.

drewblaisdell commented:
Thanks for reviewing this, everyone.

@Tug Thanks for the updates. Those changes are both great.

@stephanethomas Good catch. I replaced strings with calls to path builders where they exist. The only two I could find were the new post and plans paths, which you already linked. I'm assuming that adding a path builder for strings like `/domains/add` is out-of-scope for this PR, as that would require ~30 changes throughout the codebase, but let me know if you think we should replace them here.

I've updated the testing instructions to include the premium plan and chargeback item.

Tug commented:
I suggested the changes but @scruffian implemented it ;)

Tug commented:
Code :+1:

Tug commented:
Product :+1:
